molo/surveys/models.py

```diff
@@ -300,9 +301,10 @@ def serve_questions(self, request):
         """
         context = self.get_context(request)
         # this will only return a page if there is a translation
-        page = context['page'].get_translation_for(
-            locale=request.LANGUAGE_CODE, site=request.site)
-        if page:
+        page = get_translation_for(
+            [context['page']],
+            locale=request.LANGUAGE_CODE, site=request.site)[0]
```

Comment:
Can this return an empty list? If so then we need to check for that before trying to access one of the indexes

Reply:
yeah it can if show only translated is True and there is not translation